### ISSUE TITLE
The task of 3D object detection

### ISSUE THREAD

**Issue by AIshenfeng** [dingmyu/HR-NAS#3]
Could you release the code to support the task of 3D detection on KITTI ?
Thank you very much if you can.

**Comment by dingmyu** [comment]
Hi,

The 3D detection is based on [OpenPCDet](https://github.com/open-mmlab/OpenPCDet). Currently, the code is not organized into the HR-NAS repo. You can migrate our super network and loss functions to OpenPCDet for training.

We will support 3D detection in this repo as soon as possible, thank you.

Regards

**Comment by AIshenfeng** [comment]
First, Thanks for your reply,  and I'm looking forward to your code about  3D detection.

However, I have another question. Because I saw in the paper that "FLOPs is calculated for 2D **RPN** network using an input size of 496×432", I would like to ask whether the 3D detection task uses SSD Detection Head when searching.

Regards

**Comment by dingmyu** [comment]
Hi,

Yes, we use a single-scale anchor-based detection head, following PointPillar. See the [config](https://github.com/open-mmlab/OpenPCDet/blob/master/tools/cfgs/kitti_models/pointpillar.yaml) for details.

Regards,
Mingyu

**Comment by AIshenfeng** [comment]
That is great.

`criterion = torch.nn.CrossEntropyLoss(reduction='mean').cuda()
    criterion_smooth = optim.CrossEntropyLabelSmooth(
        FLAGS.model_kwparams['num_classes'],
        FLAGS['label_smoothing'],
        reduction='mean').cuda()
    if model.task == 'segmentation':
        criterion = CrossEntropyLoss().cuda()
        criterion_smooth = CrossEntropyLoss().cuda()
    if FLAGS.dataset == 'coco':
        criterion = JointsMSELoss(use_target_weight=True).cuda()
        criterion_smooth = JointsMSELoss(use_target_weight=True).cuda()`
It seems that there is not a loss for KITTI dataset, and not  a yml file for KITTI in config.
Could you add them?

Thanks a lot!

**Comment by AIshenfeng** [comment]
That is great.
```python：
    criterion = torch.nn.CrossEntropyLoss(reduction='mean').cuda()
    criterion_smooth = optim.CrossEntropyLabelSmooth(
        FLAGS.model_kwparams['num_classes'],
        FLAGS['label_smoothing'],
        reduction='mean').cuda()
    if model.task == 'segmentation':
        criterion = CrossEntropyLoss().cuda()
        criterion_smooth = CrossEntropyLoss().cuda()
    if FLAGS.dataset == 'coco':
        criterion = JointsMSELoss(use_target_weight=True).cuda()
        criterion_smooth = JointsMSELoss(use_target_weight=True).cuda()
 ```
It seems that there is not a loss for KITTI dataset in train.py, and not a yml file for KITTI in config.
Could you add them?

Thanks a lot!

**Comment by dingmyu** [comment]
Hi,

For 3D detection, we follow all settings in PointPillar (OpenPCDet reproduced version), the loss function of 3D detection can be found [here](https://github.com/open-mmlab/OpenPCDet/blob/master/pcdet/utils/loss_utils.py).

Thanks

**Comment by AIshenfeng** [comment]
Hi，

Your work is very good and very helpful to my research.

I try to implement 3D detection myself, but I am not clear about many settings and it is a bit difficult.

Can you tell me how to combine your code with OpenPCDet? For example, how to prune the supernet and some search-related configurations.

Of course, it’s best if you can open source the code in the short term. If so, can you tell me how long you plan to open source the code for 3D object detection?

Thank you very much!

**Comment by dingmyu** [comment]
Hi,

Thanks for your interest. I will merge the codes as soon as time and GPU resources are available, hopefully within a month.

Regards

**Comment by dingmyu** [comment]
Feel free to reopen the issue if you have any further questions.

**Comment by AIshenfeng** [comment]
Hi!

Can the code be open source as scheduled？

Very much looking forward to it~~~

best regards
